jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/FilterWidget/FilterWidget.js

```diff
@@ -0,0 +1,101 @@
+import {ActionURL} from "@labkey/api";
+import {style as styles} from "./style";
+import {filterMap as FILTERS } from "./filters"
+
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  Button,
+  List,
+  ListItem,
+  DialogActions,
+  Checkbox,
+  FormGroup,
+  FormControlLabel,
+  Tooltip,
+  IconButton,
+  AddIcon,
+  FormLabel
+} from '@material-ui/core'
+import { getContainingTrack, getSession, getContainingView, getContainingDisplay } from '@jbrowse/core/util'
+import { getConf, readConfObject } from '@jbrowse/core/configuration'
+
+
+
+export default jbrowse => {
+    const {
+        Paper,
+        Table,
+        TableBody,
+        TableCell,
+        TableHead,
+        TableRow
+    } = jbrowse.jbrequire('@material-ui/core')
+    const { observer, PropTypes: MobxPropTypes } = jbrowse.jbrequire('mobx-react')
+    const React = jbrowse.jbrequire('react')
+    const { useState, useEffect } = React
+    const { FeatureDetails, BaseCard } = jbrowse.jbrequire(
+            '@jbrowse/core/BaseFeatureWidget/BaseFeatureDetail',
+    )
+
+    function FilterForm(props){
+        const { model } = props
+        const { trackConfig } = model
+
+        //Load the defaults from the track's config. This should let us provide them at load time in the session JSON
+        const existingFilters = (readConfObject(trackConfig, ['adapter', 'filters']) || '').split(';')
+
+        // TODO: we can leave this alone for now, but once we support filters with user-supplied thresholds, this logic doesnt work anymore
+        // We should instead make this UI has 'Add' and 'Remove' buttons. The user could hit add, and then pick a filter type (based on register types defined in FILTERS).
+        let filterState = {}
+        Object.entries(FILTERS).map(([key, val]) => filterState[key] = existingFilters.indexOf(key) > -1)
+
+        const [state, setState] = React.useState(filterState)
+        const classes = styles()
+        const handleSubmit = (event) => {
+            event.preventDefault();
+            let activeFilters = []
+            Object.entries(state).map(([key, val]) => {
+                if (val)
+                    activeFilters.push(key)
+            })
+
+            //TODO: unsure if this could ever happen since we query it above?
+            if (!trackConfig.adapter.filters) {
+                console.error("trackConfig.filters was null in FilterWidget!")
+            }
+
+            // TODO: by moving the config property from the adapter to the track itself, this
+            // avoids the problem where variants stop rendering. However, this set() isnt triggering
+            // the afterAttach event in model.js.
+            trackConfig.adapter.filters.set(activeFilters.join(';'))
```

Comment:
@jctrotter can you please check out this line specifically? when I run this, executing this line seems to cause all variants to disappear. so something is going wrong in the view refresh lifecycle.